### PR TITLE
Weekly availability times

### PR DESCRIPTION
**Pull request by Peyton-McKee**
## Changes

Add Support for having multiple availabilities for different weeks
Add custom migration to migrate over old availabiliities to new SQL
no longer force people to enter in zoom link and gmail when editing availability
allow users to update their confirmed availability after theyve already confirmed it

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: src/backend/src/prisma/schema.prisma [hunk 1]
```diff
@@ -735,14 +735,23 @@ model Design_Review {
...
 }
 
+model Availability {
+  availabilityId     String            @id @default(uuid())
+  scheduleSettingsId String
+  scheduleSettings   Schedule_Settings @relation(fields: [scheduleSettingsId], references: [drScheduleSettingsId])
+
+  // Availibilies are integers between 0 and 83 from 10am - 10pm (Monday - Sunday) see meetingTime field in Design_Review
+  availability Int[]
+  dateSet      DateTime @default(now())
```
Comment (RChandler234) [inline src/backend/src/prisma/schema.prisma:745]: Is there a specific reason this is named dateSet? I feel like weekDate or smthn would be more accurate

**Review by RChandler234 — CHANGES_REQUESTED** [review]
I think there's a ticket for not forcing ppl to enter their zoom link/ stuff fyi, so u can close that since you're doing it here

File: src/backend/src/prisma/schema.prisma (hunk 1 above)
Comment (Peyton-McKee) [reply src/backend/src/prisma/schema.prisma:745]: yeah i wanna keep it more vague for when we dont just have it week by week but ideally have it the next seven days from the day you select

**Review by walker-sean — APPROVED** [review]
This is sick

**Pull request merged**